analyzers/src/SonarAnalyzer.Common/Rules/SymbolicExecutionRunnerBase.cs

```diff
@@ -103,27 +108,46 @@ private void AnalyzeRoslyn(SonarAnalysisContext sonarContext, SyntaxNodeAnalysis
 
         protected class RuleFactory
         {
-            public Type Type { get; }
             private readonly Func<SymbolicRuleCheck> createInstance;
+            private readonly Func<object> sonarFallbackFactory;
+
+            public Type Type { get; }
 
-            protected RuleFactory(Type type, Func<SymbolicRuleCheck> createInstance)
+            protected RuleFactory(Type type, Func<SymbolicRuleCheck> createInstance, Func<object> sonarFallbackFactory)
             {
                 Type = type;
                 this.createInstance = createInstance;
+                this.sonarFallbackFactory = sonarFallbackFactory;
             }
 
-            public SymbolicRuleCheck CreateInstance(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext)
+            public SymbolicRuleCheck CreateInstance(IAnalyzerConfiguration configuration, SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext)
             {
+                if (configuration.ForceSonarCfg && sonarFallbackFactory != null)
+                {
+                    return null;
+                }
+
                 var ret = createInstance();
                 ret.Init(sonarContext, nodeContext);
                 return ret;
             }
+
+            public object CreateSonarFallback(IAnalyzerConfiguration configuration) =>
+                configuration.ForceSonarCfg ? sonarFallbackFactory?.Invoke() : null;
+        }
+
+        protected class RuleFactory<TCheck> : RuleFactory
+            where TCheck : SymbolicRuleCheck, new()
+        {
+            public RuleFactory() : this(null) { }
+            public RuleFactory(Func<object> sonarFallbackFactory) : base(typeof(TCheck), () => new TCheck(), sonarFallbackFactory) { }
```

Comment:
This one should be protected for the system to make sense, right? 
```suggestion
            protected RuleFactory(Func<object> sonarFallbackFactory) : base(typeof(TCheck), () => new TCheck(), sonarFallbackFactory) { }
```

Reply:
protected gives a compiler error (private protected too) on the ctor. I changed the class to private sealed and made the ctors internal.

Reply:
The `protected` comment was actually outdated after the next one below, because the this pass-through constructor was removed.

Reply:
`protected` class and `public` constructor is the desired state

Reply:
Just to make sure: `private sealed class` was the original state. I changed it to `protected sealed`now.